src/inmanta/protocol/rest/__init__.py

```diff
@@ -218,6 +238,38 @@ async def process(self) -> None:
 
         self._processed = True
 
+    async def _get_dict_value_from_message(
+        self, arg: str, dict_prefix: str, dict_with_prefixed_names: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        value = {k[len(dict_prefix) :]: v for k, v in dict_with_prefixed_names.items()}
+        # Check if the values should be converted to lists
+        type_args = typing_inspect.get_args(self._argspec.annotations.get(arg), evaluate=True)
+        if issubclass(type_args[1], list):
+            value = {
+                key: self._get_list_value_from_url(val) if not isinstance(val, list) else val for key, val in value.items()
+            }
+        return value
+
+    def _is_dict_or_optional_dict(self, arg_type: Type) -> bool:
+        if typing_inspect.is_optional_type(arg_type):
+            arg_type = typing_inspect.get_args(arg_type, evaluate=True)[0]
+        return issubclass(arg_type, dict)
+
+    def _get_list_value_from_url(self, value: str) -> List[str]:
+        # Split by the delimiter
+        split_parts = value.split(",")
+        split_parts = [part.replace("%2C", ",") for part in split_parts]
```

Comment:
What is the argument actually contains '%2C', do we get an extra argument?

Reply:
@wouterdb it turns out that simply adding the list parameter multiple times to a query string to get a list has much better support, so I ended up using that one. So instead of `param=value1,value2` just `param=value1&param=value2`. 
I'll update the design document as well